mpegts/mpegts_demuxer.cpp

```diff
@@ -97,85 +97,87 @@ uint8_t MpegTsDemuxer::decode(SimpleBuffer &rIn) {
 
             if (lTsHeader.mAdaptationFieldControl == MpegTsAdaptationFieldType::mPayloadOnly ||
                 lTsHeader.mAdaptationFieldControl == MpegTsAdaptationFieldType::mPayloadAdaptionBoth) {
-                PESHeader lPesHeader;
-                if (lTsHeader.mPayloadUnitStartIndicator == 0x01) {
 
-                    mEsFrames[lTsHeader.mPid]->mRandomAccess = lRandomAccessIndicator;
+                EsFrame* lEsFrame = mEsFrames[lTsHeader.mPid].get();
+                if (lTsHeader.mPayloadUnitStartIndicator == 0x01) {

```

Comment:
This block to flush out the current frame, before assigning next frames rai is the actual fix in this PR, rest is cosmetics :P
So moving `mEsFrames[lTsHeader.mPid]->mRandomAccess = lRandomAccessIndicator` to after this block.